pycytominer/cyto_utils/cells.py

```diff
@@ -244,11 +256,33 @@ def load_image(self):
 
         """
 
-        # Extract image metadata
-        image_query = "select {} from image".format(
-            ", ".join(np.union1d(self.image_cols, self.strata))
-        )
+        image_query = "select * from image"
         self.image_df = pd.read_sql(sql=image_query, con=self.conn)
+
+        if self.add_image_features:
+            check_image_features(
+                self.image_feature_categories, list(self.image_df.columns)
+            )
+
+            image_features = list(
+                self.image_df.columns[
+                    self.image_df.columns.str.startswith(
+                        tuple(self.image_feature_categories)
+                    )
+                ]
+            )
```

Comment:
This is a bit confusing to me. The concern is that a user might provide image features (that MUST be capitalized) (based on line 131) that aren't in `image_df`?

Without reading `check_image_features`, I am wondering if that function should also handle the output the compatible user input image_features.

Doing so would do two things:

1. Make this function more readable (and reduce length!)
2. Enable us to more easily update how a user would interact with the image_df (which would help with maintainability)

Just a thought! I might not be seeing all the constraints tho

Reply:
You are right, this is confusing to me as well. Will reconfigure it.

Reply:
Refactored the image feature extraction part of the code. I hope this makes more sense - https://github.com/cytomining/pycytominer/pull/168/commits/67912d4c5881f028994ec5405c3ff32357f799bb